lib/internal/fs/dir.js

```diff
@@ -157,16 +158,24 @@ class Dir {
   }
 
   close(callback) {
-    if (this[kDirClosed] === true) {
-      throw new ERR_DIR_CLOSED();
-    }
-
+    // Promise
     if (callback === undefined) {
+      if (this[kDirClosed] === true) {
+        return new Promise((_, reject) => reject(new ERR_DIR_CLOSED()));
+      }
       return this[kDirClosePromisified]();
-    } else if (typeof callback !== 'function') {
+    }
+
+    // callback
+    if (typeof callback !== 'function') {
       throw new ERR_INVALID_CALLBACK(callback);
     }
 
+    if (this[kDirClosed] === true) {
+      callback(new ERR_DIR_CLOSED());
```

Comment:
I'm not sure, but maybe this should be called on the next tick?

Reply:
Thanks for the feedback, errors in asynchronous functions should be executed in nextTick, but I'm not sure if this is necessary.

Is there a specification for error handling of asynchronous functions that I can refer to?

Reply:
It's important not to break user expectations. Sending the error synchronously would change the order of operations:
```js
console.log(1);
dir.close(err=>console.log(3));
console.log(2); // This should be executed before the callback
```

Reply:
```suggestion
      process.nextTick(callback, new ERR_DIR_CLOSED());
```